orbit/cmd/desktop/desktop_linux.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"slices"
+
+	"github.com/godbus/dbus/v5"
+	"github.com/rs/zerolog/log"
+)
+
+//go:embed icon_dark.png
+var iconDark []byte
+
+func blockWaitForStopEvent(_ string) error {
+	log.Debug().Msg("communication channel helpers are not implemented for this platform")
+	return nil
+}
+
+func trayIconExists() bool {
+	conn, err := dbus.SessionBus()
+	if err != nil {
+		log.Error().Err(err)
+	}
+
+	// Get the name we would expect systray to reserve for our tray icon.
+	trayIconDbusName := fmt.Sprintf("org.kde.StatusNotifierItem-%d-1", os.Getpid())
```

Comment:
This is the most straightforward approach I could think of.  My one hesitation is that `systray` doesn't consider it a fatal error if it can't reserve this name.  So in the edge case that the name never gets reserved, we would kill the process every time.  I'm not sure this is a real-world case though -- my instinct is that if systray can't reserve the name, it's because another instance in the same process has already done so, in which case this code still works fine.

Reply:
Also this relies somewhat on the internals of systray to determine the name, but it seems like the standard way of formatting dbus names.